docs/naming-conventions.md

```diff
@@ -0,0 +1,44 @@
+# Naming Conventions
+
+All names in this project should adhere to the guidelines outlined in this document.
+
+## Rust Code
+
+All variable names, function names, module names, and macros in Rust code should follow typical `snake_case` conventions. All Rust types, traits, structs, and enum variants must follow `UpperCamelCase`. Additionally, static and constant variables should be written in `SCREAMING_SNAKE_CASE`. For the complete list of Rust naming conventions, see the [Rust Style Guide](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html).
+
+As a final rule, the functions of all component `ffi/src/lib.rs` files should also follow *snake_case*, but with an additional prefix based on the library of that function. 
+
+Examples:
+```
+	sync15_passwords_get_all()
+	places_query_autocomplete()
+	fxa_handle_push_message()
+```
+
+## Swift Code
+
+All code written in Swift should follow [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines/). Names of types and protocols are `UpperCamelCase`, while all other uses are `lowerCamelCase`.  
+
+The only exception to this rule for this project is to use Rust convention for FFI binding files (e.g. `RustFxAFFI.h`).
+
+## Kotlin Code
+
+If a source file contains only a top-level class, the source file should reflect the case-sensitive name of the class plus the *.kt* extension. Otherwise, if the source contains multiple top-level declarations, choose a name that describes the contents of the file, apply `UpperCamelCase` and append `.kt` extension.
+
+Examples:
+
+```
+	//FooBar.kt
+	class FooBar{}
+```
+
+### Naming Rules
```

Comment:
Make formatting consistent between the three. Rules, link to style guide, examples.